libs/ledger/tests/consensus/pos_tests.cpp

```diff
@@ -0,0 +1,258 @@
+  Members                                     cabinet_;  // The identities that are in the cabinet
+  Members                                     qual_;     // The identities that are in qual
+  StakeManagerPtr                             stake_;
+  BeaconSetupServicePtr                       beacon_setup_;
+  BeaconServicePtr                            beacon_;
+  MainChain                                   chain_;
+  StorageInterface                            storage_;
+  Identity                                    mining_identity_;
+  uint64_t const                              aeon_period_{10};
+  uint64_t const                              max_cabinet_size_{10};
+  uint64_t const                              block_interval_ms_{5000};
+  NotarisationPtr                             notarisation_;
+};
+
+TEST_F(ConsensusTests, test_valid_block)
+{
+  BlockPtr block = ValidNthBlock(1);
+
+  ASSERT_EQ(consensus_->ValidBlock(*block), ledger::ConsensusInterface::Status::YES);
+}
+
+TEST_F(ConsensusTests, test_loose_block)
+{
+  BlockPtr block = ValidNthBlock(1);
+  block->block_number++;
+
+  ASSERT_EQ(consensus_->ValidBlock(*block), ledger::ConsensusInterface::Status::NO);
+}
+
+TEST_F(ConsensusTests, test_entropy_block_number_mismatch)
+{
+  BlockPtr block = ValidNthBlock(1);
+  block->block_entropy.block_number++;
+
+  ASSERT_EQ(consensus_->ValidBlock(*block), ledger::ConsensusInterface::Status::NO);
+}
+
+TEST_F(ConsensusTests, test_hash_empty)
+{
+  BlockPtr block = ValidNthBlock(1);
+  block->hash    = "";
+
+  ASSERT_EQ(consensus_->ValidBlock(*block), ledger::ConsensusInterface::Status::NO);
+}
+
+TEST_F(ConsensusTests, test_wrong_miner_identity)
+{
+  BlockPtr block  = ValidNthBlock(1);
+  block->miner_id = qual_[1];
+
+  ASSERT_EQ(consensus_->ValidBlock(*block), ledger::ConsensusInterface::Status::NO);
+}
+
+TEST_F(ConsensusTests, test_not_an_aeon)
+{
+  BlockPtr block = ValidNthBlock(1);
+  block->block_entropy.confirmations.clear();
+
+  ASSERT_EQ(consensus_->ValidBlock(*block), ledger::ConsensusInterface::Status::NO);
+}
+
+TEST_F(ConsensusTests, DISABLED_test_zero_weight)
+{
+  BlockPtr block = ValidNthBlock(1);
+  block->weight  = 0;
+
+  ASSERT_EQ(consensus_->ValidBlock(*block), ledger::ConsensusInterface::Status::NO);
+}
+
+TEST_F(ConsensusTests, DISABLED_test_wrong_weight)
+{
+  BlockPtr block = ValidNthBlock(1);
+  block->weight++;
+
+  ASSERT_EQ(consensus_->ValidBlock(*block), ledger::ConsensusInterface::Status::NO);
+}
+
+TEST_F(ConsensusTests, test_timestamp_too_early)
+{
+  BlockPtr block = ValidNthBlock(1);
+  block          = ValidNthBlock(2);
+
+  ASSERT_EQ(consensus_->ValidBlock(*block), ledger::ConsensusInterface::Status::NO);
+}
+
+TEST_F(ConsensusTests, test_timestamp_before_previous)
+{
+  BlockPtr block = ValidNthBlock(1);
+  block          = ValidNthBlock(2);
+
+  block->timestamp = 0;
+
+  ASSERT_EQ(consensus_->ValidBlock(*block), ledger::ConsensusInterface::Status::NO);
```

Comment:
Same here, doesn't block at block number 1 have to be added to consensus first?

Reply:
as comment above